databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -180,18 +159,193 @@ def update_share_cli(api_client, name, new_name, comment, owner,
             raise ValueError('Cannot specify JSON if any other update flags are specified')
         updates = []
         for a in add_table:
-            updates.append({'action': 'ADD', 'data_object': shared_data_object(a)})
+            updates.append({'action': 'ADD', 'data_object': shared_table_object(a)})
         for r in remove_table:
-            updates.append({'action': 'REMOVE', 'data_object': shared_data_object(r)})
+            updates.append({'action': 'REMOVE', 'data_object': shared_table_object(r)})
         data = {'name': new_name, 'comment': comment, 'owner': owner, 'updates': updates}
         share_json = UnityCatalogApi(api_client).update_share(name, data)
         click.echo(mc_pretty_format(share_json))
     else:
         json_cli_base(json_file, json,
                       lambda json: UnityCatalogApi(api_client).update_share(name, json))
 
+def shared_schema_object(name=None, comment=None):
+    val = {
+        'data_object_type': 'SCHEMA'
+    }
+    if name is not None:
+        val['name'] = name
+    if comment is not None:
+        val['comment'] = comment
+    return val
+
+def create_common_shared_schema_options(f):
+    @click.option('--schema', required=True,
+                  help='Full name of the shared schema.')
+    @click.option('--comment', default=None,
+                  help='New comment of the shared schema.')
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        f(*args, **kwargs)
+    return wrapper
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Add a shared schema.')
+@click.option('--share', required=True,
+              help='Name of the share to update.')
+@create_common_shared_schema_options
+@click.option('--json-file', default=None, type=click.Path(),
+              help="Adds a shared table based on shared data object represented in JSON file.")
```

Comment:
These JSON options have the same functionality for adding any object type. You should add validation on the type to ensure it matches the object type.

Same with the add table CLI. Since we only have table as the type today, it should not have any backwards compatibility issues.

Reply:
Fixed